frontend/src/views/AnnouncementView/AnnouncementView.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { AnnouncementBrowser, HeadingButton } from '../../components';
+import { HeadingSection, StyledH1 } from 'commonStyles';
+import { RootState } from '../../types';
+import ac, { Resources } from '../../security/accessControl';
+import penWhiteIcon from 'assets/icons/penWhite.svg';
+
+const AnnouncementView = (): JSX.Element => {
+  const userRole = useSelector((state: RootState) => state.user.role);
+  const [isAbleToModify, setIsAbleToModify] = useState(false);
+
+  useEffect(() => {
+    setIsAbleToModify(checkIfAbleToModify());
+  }, [userRole]);
+
+  const checkIfAbleToModify = (): boolean => {
+    if (userRole !== null) {
+      const permissions = ac.can(userRole).createAny(Resources.Announcements)
+        .attributes;
+      return permissions.includes('*');
```

Comment:
For AccessControl, as long as you run `ac.can(userRole).createAny(Resources.Announcements).granted`, you will get a boolean that you can compare instead of checking all attributes.